compiler/rustc_query_system/src/ich/impls_syntax.rs

```diff
@@ -80,9 +80,16 @@ impl<'a> HashStable<StableHashingContext<'a>> for SourceFile {
         src_hash.hash_stable(hcx, hasher);
 
         // We only hash the relative position within this source_file
-        lines.len().hash_stable(hcx, hasher);
-        for &line in lines.iter() {
-            stable_byte_pos(line, start_pos).hash_stable(hcx, hasher);
+        match &*lines.borrow() {
+            SourceFileLines::Lines { lines } => {
+                lines.len().hash_stable(hcx, hasher);
+                for &line in lines.iter() {
+                    stable_byte_pos(line, start_pos).hash_stable(hcx, hasher);
+                }
+            }
+            SourceFileLines::Diffs { .. } => {
+                panic!("called hash_stable on SourceFileLines::Diffs for {:?}", name);
```

Comment:
Is there a reason we panic! here vs. decoding the diffs for hashing purposes? Can we leave a comment with that reason, so that if/when this panic occurs in the future there's immediate context for it?